contracts/DataProduct.sol

```diff
@@ -1,38 +1,28 @@
 pragma solidity 0.4.24;
 
 import "./AddressArrayRemover.sol";
-import "./SafeMath.sol";
+import "./DataProductInterface.sol";
 import "./ERC20.sol";
 import "./Ownable.sol";
-import "./Registry.sol";
+import "./RegistryInterface.sol";
+import "./SafeMath.sol";
+import "./TransactionInterface.sol";
+import "./TransactionFactoryInterface.sol";
 
 
-contract DataProduct is Ownable {
+contract DataProduct is Ownable, DataProductInterface {
     using AddressArrayRemover for address[];
     using SafeMath for uint256;
 
-    uint8 constant private minScore = 1;
-    uint8 constant private maxScore = 5;
-
-    struct Transaction {
-        address wallet;
-        string publicKey;
-        string buyerMetaHash;
-        uint256 rateDeadline;
-        uint256 deliveryDeadline;
-        uint256 price;
-        uint256 fee;
-        bool purchased;
-        bool finalised;
-        bool rated;
-        uint8 rating;
-    }
-
-    mapping(address => Transaction) private transactions;
+    mapping(address => address) private buyersTransactions;
     address[] private buyersAddresses;
+    address[] private transactionsAddresses;
```

Comment:
I think it would be better if you create a getTransactionsAddresses view instead of storing addresses in array. These addresses could be fetched using buyersTransactions and buyersAddresses.